src/contract/deploy.rs

```diff
@@ -120,9 +121,18 @@ impl<T: Transport> Future for PendingContract<T> {
         let eth = self.eth.take().expect("future polled after ready; qed");
         let abi = self.abi.take().expect("future polled after ready; qed");
 
-        match receipt.contract_address {
-            Some(address) => Ok(Async::Ready(Contract::new(eth, address, abi))),
-            None => Err(Error::ContractDeploymentFailure(receipt.transaction_hash)),
+        let status = receipt.status.ok_or(Error::Api(ApiError::InvalidResponse(String::from(
```

Comment:
It's not an error, for chains that don't support `status` the value will be `None`. Please just fall back to `contratAddress` detection in such case as I've suggested in the issue.